pkg/controller/compute/kubernetes/aws_handler_test.go

```diff
@@ -0,0 +1,268 @@
+		},
+	}
+	createError := errors.New("test-cluster-create-error")
+	type args struct {
+		class *corev1alpha1.ResourceClass
+		claim corev1alpha1.ResourceClaim
+		c     client.Client
+	}
+	type want struct {
+		err error
+		res corev1alpha1.Resource
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "success",
+			args: args{
+				class: class,
+				claim: claim,
+				c:     test.NewMockClient(),
+			},
+			want: want{
+				res: &v1alpha1.EKSCluster{
+					ObjectMeta: v1.ObjectMeta{
+						Labels:          map[string]string{labelProviderKey: labelProviderAWS},
+						Namespace:       class.Namespace,
+						Name:            "eks-test-claim-uid",
+						OwnerReferences: []v1.OwnerReference{claim.OwnerReference()},
+					},
+					Spec: v1alpha1.EKSClusterSpec{
+						ClassRef: class.ObjectReference(),
+						ClaimRef: claim.ObjectReference(),
+					},
+				},
+			},
+		},
+		{
+			name: "failure",
+			args: args{
+				class: class,
+				claim: claim,
+				c: &test.MockClient{
+					MockCreate: func(ctx context.Context, obj runtime.Object) error {
+						return createError
+					},
+				},
+			},
+			want: want{
+				err: errors.Wrapf(createError,
+					"failed to create cluster %s/%s", class.Namespace, "eks-"+claim.UID),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &EKSClusterHandler{}
+			got, err := r.Provision(tt.args.class, tt.args.claim, tt.args.c)
+			if diff := deep.Equal(err, tt.want.err); diff != nil {
+				t.Errorf("EKSClusterHandler.Provision() error = %v, want.err %v\n%s", err, tt.want.err, diff)
+				return
+			}
+			if diff := deep.Equal(got, tt.want.res); diff != nil {
+				t.Errorf("EKSClusterHandler.Provision() = \n%v, want.res \n%v\n%s", got, tt.want.res, diff)
+			}
+		})
+	}
+}
+
+func TestEKSClusterHandler_SetBindStatus(t *testing.T) {
+	name := types.NamespacedName{Namespace: "foo", Name: "bar"}
+
+	getError := errors.New("test-get-error")
+	getErrorNotFound := kerrors.NewNotFound(schema.GroupResource{}, name.String())
+	updateError := errors.New("test-update-error")
+
+	type args struct {
+		name  types.NamespacedName
+		c     client.Client
+		bound bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want error
+	}{
+		{
+			name: "failure",
+			args: args{
+				name: name,
+				c: &test.MockClient{
+					MockGet: func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+						return getError
+					},
+				},
+			},
+			want: errors.Wrapf(getError, "failed to retrieve cluster %s", name),
+		},
+		{
+			name: "failure - not found, bound",
+			args: args{
+				name: name,
+				c: &test.MockClient{
+					MockGet: func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+						return getErrorNotFound
+					},
+				},
+				bound: true,
+			},
+			want: errors.Wrapf(getErrorNotFound, "failed to retrieve cluster %s", name),
+		},
+		{
+			name: "failure - not found, not bound",
```

Comment:
I think `t.Run` will mangle this name into something like `failure___not_found__not_bound` when the test is run. It may be more readable to use snake or camel case test names explicitly.

Reply:
I see what you mean:
![image](https://user-images.githubusercontent.com/324803/56526173-d46f9a80-6500-11e9-9603-d8845814d563.png)

Not :100: if `failureNotFoundBound` is more readable than `failure_-_not_found,_bound`

Reply:
My two cents is `FailureNotFoundBound`, because it matches the style of 'parent' test names, which are typically public Go functions like `TestSomeCamelCaseThing`.

Reply:
Good point, @negz. I created https://github.com/crossplaneio/crossplane/issues/412 to track this change.